src/frequenz/sdk/timeseries/_serialized_ringbuffer.py

```diff
@@ -0,0 +1,77 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Ringbuffer implementation with serialization support."""
+
+import pickle
+from datetime import datetime, timedelta
+from os.path import exists
+from typing import TypeVar
+
+import numpy as np
+
+from frequenz.sdk.timeseries._ringbuffer import OrderedRingBuffer
+
+T = TypeVar("T")
+Container = TypeVar("Container", list, np.ndarray)
+
+ATTRIBUTES_TO_PICKLE = [
+    "_datetime_oldest",
+    "_datetime_newest",
+    "_buffer",
+    "_gaps",
+    "_sampling_period",
+    "_time_range",
+    "_time_index_alignment",
+]
+
+
+class SerializedRingbuffer(OrderedRingBuffer[T, Container]):
+    """Sorted ringbuffer with serialization support."""
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        buffer: Container,
+        sampling_period: timedelta,
+        path: str,
+        time_index_alignment: datetime = datetime(1, 1, 1),
+        load_from_disk: bool = True,
+    ) -> None:
+        """Initialize the time aware ringbuffer.
+
+        Args:
+            buffer: Instance of a buffer container to use internally.
+            sampling_period: Timedelta of the desired resampling period.
+            path: Path to where the data should be saved.
+            time_index_alignment: Arbitary point in time used to align
+                timestamped data with the index position in the buffer.
+                Used to make the data stored in the buffer align with the
+                beginning and end of the buffer borders.
+                For example, if the `time_index_alignment` is set to
+                "0001-01-01 12:00:00", and the `sampling_period` is set to
+                1 hour and the length of the buffer is 24, then the data
+                stored in the buffer could correspond to the time range from
+                "2022-01-01 12:00:00" to "2022-01-02 12:00:00" (date chosen
+                arbitrarily here).
+            load_from_disk: Default True, if True, will load the data from disk
+                upon start.
+        """
+        super().__init__(buffer, sampling_period, time_index_alignment)
+        self._path = path
```

Comment:
Should the `path` be checked? (path exists or write permission)
My concern here is, for instance, `load()` relies on the path to load data. I also wonder if it would be more convenient to pass to load() and dump() as a parameter instead

Reply:
hmm on the other hand it could create inconsistencies between loaded and dumped data if the path is passed as a parameter to the functions